dotcom-rendering/src/lib/useAdBlockAsk.ts

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useState } from 'react';
+import { useAB } from './useAB';
+
+const useIsInAdBlockAskVariant = (): boolean => {
+	const abTestAPI = useAB()?.api;
+	const isInVariant = !!abTestAPI?.isUserInVariant('AdBlockAsk', 'variant');
+	return isInVariant;
+};
+
+const detectByRequests = async () => {
+	/**
+	 * Make a HEAD request to a URL that is typically blocked by ad-blockers
+	 */
+	const requestDoubleclick = async () => {
```

Comment:
Would it make sense to receive the timeout as a parameter, even if using a default value of 1000?

Do these methods gain anything from being defined inside the `detectByRequests`’s body?

Reply:
No reason really other than scoping, but probably more idiomatic in DCR to have them at the top-level, especially since they don't close over any values.